src/layer/LayersFactory.ts

```diff
@@ -138,25 +138,34 @@ export class LayersFactory {
     const filteredLayersInfos =
       filterLayers === null ? layersInfos : layersInfos.filter(l => filterLayers(l.layerId, l.dataset));
 
-    return filteredLayersInfos.map(({ layerId, dataset, title, description }) => {
-      if (!dataset) {
-        return new WmsLayer({ baseUrl, layerId, title, description });
-      }
+    return Promise.all(
+      filteredLayersInfos.map(async ({ layerId, dataset, title, description }) => {
+        if (!dataset) {
+          return new WmsLayer({ baseUrl, layerId, title, description });
+        }
 
-      const SHLayerClass = LayersFactory.LAYER_FROM_DATASET_V3[dataset.id];
-      if (!SHLayerClass) {
-        throw new Error(`Dataset ${dataset.id} is not defined in LayersFactory.LAYER_FROM_DATASET`);
-      }
-      return new SHLayerClass({
-        instanceId: parseSHInstanceId(baseUrl),
-        layerId,
-        evalscript: null,
-        evalscriptUrl: null,
-        dataProduct: null,
-        title,
-        description,
-      });
-    });
+        const SHLayerClass = LayersFactory.LAYER_FROM_DATASET_V3[dataset.id];
+        if (!SHLayerClass) {
+          throw new Error(`Dataset ${dataset.id} is not defined in LayersFactory.LAYER_FROM_DATASET`);
+        }
+
+        const layer = new SHLayerClass({
+          instanceId: parseSHInstanceId(baseUrl),
+          layerId,
+          evalscript: null,
+          evalscriptUrl: null,
+          dataProduct: null,
+          title,
+          description,
+        });
+
+        if (layer instanceof BYOCLayer) {
+          await layer.updateLayerFromServiceIfNeeded();
+        }
```

Comment:
I understand the reasoning behind updating the layer here, but the convention elsewhere is that we lazy-load the needed data only when needed. In other words, every method in `BYOCLayer` that needs a `collectionId` should perform `updateLayerFromServiceIfNeeded` (actually, I thought it already did?).

Also, while I didn't test it, I think constructing a BYOCLayer with only `instanceId` and `layerId` would not work then?
```
  const layer = new BYOCLayer({ instanceId: ..., layerId: ... });
```

I would suggest using lazy-loading when needed, instead of on construction...